client/my-sites/themes/auto-loading-homepage-modal.jsx

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { Dialog } from '@automattic/components';
+import {
+	getCanonicalTheme,
+	hasActivatedTheme,
+	themeHasAutoLoadingHomepage,
+	isActivatingTheme,
+	isThemeActive,
+	shouldShowHomepageWarning,
+	getPreActivateThemeId,
+} from 'state/themes/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	acceptAutoLoadingHomepageWarning,
+	hideAutoLoadingHomepageWarning,
+	activate as activateTheme,
+} from 'state/themes/actions';
+
+/**
+ * Style dependencies
+ */
+import './auto-loading-homepage-modal.scss';
+
+class AutoLoadingHomepageModal extends Component {
+	static propTypes = {
+		source: PropTypes.oneOf( [ 'details', 'list', 'upload' ] ).isRequired,
+		theme: PropTypes.shape( {
+			author: PropTypes.string,
+			author_uri: PropTypes.string,
+			id: PropTypes.string,
+			name: PropTypes.string,
+		} ),
+		hasActivated: PropTypes.bool.isRequired,
+		isActivating: PropTypes.bool.isRequired,
+		hasAutoLoadingHomepage: PropTypes.bool,
+		siteId: PropTypes.number,
+		isVisible: PropTypes.bool,
+		onClose: PropTypes.func,
+		installingThemeId: PropTypes.string.isRequired,
+	};
+
+	closeModalHandler = ( activate = false ) => () => {
+		if ( activate ) {
+			const { installingThemeId, siteId, source } = this.props;
+			this.props.acceptAutoLoadingHomepageWarning( installingThemeId );
+			return this.props.activateTheme( installingThemeId, siteId, source );
+		}
+		this.props.hideAutoLoadingHomepageWarning();
+	};
+
+	render() {
+		const {
+			theme,
+			hasActivated,
+			isActivating,
+			hasAutoLoadingHomepage,
+			isCurrentTheme,
+			isVisible = false,
+		} = this.props;
+
+		// Nothing to do when it's the current theme.
+		if ( isCurrentTheme ) {
+			return null;
+		}
+
+		// Nothing to show if the theme doesn't have auto loading homepage.
+		if ( ! hasAutoLoadingHomepage ) {
+			return null;
+		}
+
+		// Hide while is activating or when it's activated.
+		if ( isActivating || hasActivated ) {
+			return null;
+		}
+
+		if ( ! theme ) {
+			return null;
+		}
+
+		const { name: themeName } = this.props.theme;
+
+		return (
+			<Dialog
+				className="themes__auto-loading-homepage-modal"
+				isVisible={ isVisible }
+				buttons={ [
+					{
+						action: 'keepCurrentTheme',
+						label: translate( 'No, keep my current theme' ),
+						isPrimary: false,
+						onClick: this.closeModalHandler( false ),
+					},
+					{
+						action: 'activeTheme',
+						label: translate( 'Yes, active %(themeName)s', {
```

Comment:
Minor but should this read `activate`?

Reply:
Nice catch @frontdevde !

Reply:
Thanks, Michael.
I've just copied what the design defines in https://github.com/Automattic/wp-calypso/issues/39073.

<img src="https://user-images.githubusercontent.com/1382943/73408469-511e7900-42ca-11ea-983a-bd49a2e93007.jpg" />

Anyway, updating...

Reply:
[Fixed here](https://github.com/Automattic/wp-calypso/pull/39436/commits/54e4f73d7d9c9b82b97936ff3e3f5e0bf4e0ead1).
![image](https://user-images.githubusercontent.com/77539/74846364-18911e80-530f-11ea-8f66-4cb261f09c33.png)

Thanks!